### PR TITLE
LRS Demo Cloudformation Templates

### PR DESCRIPTION
**Pull request by cliffcaseyyet**
- VPC
- Postgres DB
- Credentials Init for PG DB using Custom Resource Lambda
- App Server, with service def, version param, etc
- LB
- SSL/Cert and Domain Updates

### CODE REVIEW

File: dev-resources/template/2_lrs.yml [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+  # DB Details
+  DBName:
+    Description: PG Database name. Ignored if DBSnapshotIdentifier is provided
+    Type: String
+    Default: 'lrsql_demo'
+  DBHost:
+    Description: Write hostname of PG db
+    Type: String
+  DBPort:
+    Description: Write hostname of PG db
```
Comment (milt) [inline dev-resources/template/2_lrs.yml:20]: copypasta... port

File: dev-resources/template/2_lrs.yml [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+      VpcId: !Ref VPCId
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: '73.133.206.211/32'
+        - IpProtocol: tcp
+          FromPort: 8080
+          ToPort: 8080
+          CidrIp: '73.133.206.211/32'
```
Comment (milt) [inline dev-resources/template/2_lrs.yml:262]: parameterize

File: dev-resources/template/2_lrs.yml [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+                Type=simple
+                ExecStart=/opt/aws/bin/cfn-hup
+                Restart=always
+                [Install]
+                WantedBy=multi-user.target
+            '/etc/cfn/hooks.d/tc-auto-reloader.conf':
+              content: !Sub |
+                [cfn-auto-reloader-hook]
+                triggers=post.update
+                path=Resources.BastionInstance.Metadata.AWS::CloudFormation::Init.02_configureLrs
```
Comment (milt) [inline dev-resources/template/2_lrs.yml:337]: This is LrsInstance but this auto-reloader is trying to use `BastionInstance` from TC. It won't work so good!

**Review by milt — COMMENTED** [review]
Strategy looks good, a few minor cfn-init things with notes inline

File: dev-resources/template/2_lrs.yml (hunk 1 above)
Comment (cliffcaseyyet) [reply dev-resources/template/2_lrs.yml]: done

File: dev-resources/template/2_lrs.yml (hunk 2 above)
Comment (cliffcaseyyet) [reply dev-resources/template/2_lrs.yml]: done

File: dev-resources/template/2_lrs.yml [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+        - IpProtocol: tcp
+          FromPort: 8080
+          ToPort: 8080
+          CidrIp: '73.133.206.211/32'
+
+
+  DBInstanceIngress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      Description: Ingress from the Bastion instance to RDS instance
```
[the remark this answers is not in the document]
Comment (cliffcaseyyet) [reply dev-resources/template/2_lrs.yml]: done

File: dev-resources/template/2_lrs.yml (hunk 3 above)
Comment (cliffcaseyyet) [reply dev-resources/template/2_lrs.yml]: done - thought i got all of those!

File: dev-resources/template/2_lrs.yml [hunk 5]
```diff
@@ -0,0 +1,536 @@
...
+          commands:
+            01start_tc:
+              command: !Sub |
+                systemctl start lrsql.service
+        05_restartLrs:
+          commands:
+            01daemon_reload:
+              command: |
+                systemctl daemon-reload
+            02start_tc:
```
[the remark this answers is not in the document]
Comment (cliffcaseyyet) [reply dev-resources/template/2_lrs.yml]: done

File: dev-resources/template/db-init-script/index.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
+const { Client } = require('pg')
+var pgformat = require('pg-format');
+const AWS = require("aws-sdk");
+const sm = new AWS.SSM();
+const cfnr = require('./cfn-response.js');
+
+//helper to grab and parse secure strings from ssm
+const getParam = async (path, secure) => {
+   try {
+        const appPassword = await sm.getParameter({
```
Comment (cliffcaseyyet) [inline dev-resources/template/db-init-script/index.js:10]: kelvin says: change to appParam

**Review by milt — CHANGES_REQUESTED** [review]
Keyname should be optional, for nice no-access instances

**Comment by milt** [comment]
Consider putting an output on the LRS template with the uri of the admin interface!

**Comment by cliffcaseyyet** [comment]
> Consider putting an output on the LRS template with the uri of the admin interface!

So I did consider it. The reason i didn't is its literally directly an input of the stack and it felt weird to spit back a param. I can do it though i guess it could be handy.

**Comment by cliffcaseyyet** [comment]
> Keyname should be optional, for nice no-access instances

this is done.

**Comment by cliffcaseyyet** [comment]
> Consider putting an output on the LRS template with the uri of the admin interface!

done!

**Review by milt — APPROVED** [review]
Looks good!

**Review by blakeplock — APPROVED** [review]
Might be good practice to add https://www.sqllrs.com/ to the top level descriptions of any of these projects. Otherwise, ready to go.

**Pull request merged**